lib/neo4j/active_node/persistence.rb

```diff
@@ -128,17 +128,14 @@ def create!(*args)
         end
       end
 
-      def merge(attributes)
-        neo4j_session.query.merge(n: {self.mapped_label_names => attributes})
-          .on_create_set(n: on_create_props(attributes))
-          .on_match_set(n: on_match_props)
+      def merge(match_attributes, update_attributes = {})
+        find_or_create_query(match_attributes, update_attributes)
+          .on_match_set(n: on_match_props(update_attributes))
```

Comment:
If I'm reading it right, the `match_attributes` is used to determine which attributes are `MATCH`ed as part of the `MERGE`, but there's nothing that gives the user control over the `ON MATCH SET`